## Get Pubchem drug feautures

1. finding corresponding Pubchem ids for the drugs 
2. call Pubchem to get chemical properties of the drugs
3. Preprocess text Drug description from the original datasets
4. Preprocess some text characteristics from PubChem properties

In [ ]:
pip install pubchempy

Looking in indexes: https://pypi.org/simple, https://us-python.pkg.dev/colab-wheels/public/simple/


In [ ]:
from google.colab import drive
drive.mount('/content/gdrive')
import sys
sys.path.append('/content/gdrive/MyDrive/3rd-Year-Project/DrugProfiles_2')

Drive already mounted at /content/gdrive; to attempt to forcibly remount, call drive.mount("/content/gdrive", force_remount=True).


In [ ]:
!ls
!cd /content/gdrive/MyDrive/3rd-Year-Project/DrugProfiles_2

gdrive	sample_data


In [ ]:



import pandas as pd
import numpy as np
import os
# pip install PubChemPy
import pubchempy as pcp
import re
from pubchempy import Compound
import warnings
warnings.filterwarnings("ignore")
import time
from tqdm import tqdm
import pickle
from functions.drug_features import *


_FOLDER = "/content/gdrive/MyDrive/3rd-Year-Project/DrugProfiles_2/database/"
_FOLDER_2 = "/content/gdrive/MyDrive/3rd-Year-Project/DrugProfiles_2/results/"

In [62]:
drug_features = pd.read_csv(_FOLDER + "Drug_Features.csv").rename(columns={"Drug ID": "DRUG_ID", 
                                                                           "Drug Name": "Drug_Name",
                                                                          "Target Pathway": "Target_Pathway"})
#drug_features.set_index("DRUG_ID", inplace= True)
df = pd.read_csv(_FOLDER+'Drug_Features.csv')
ind_wrong_name = df[df["Drug Name"]=="Lestauritinib"].index
df.loc[ind_wrong_name, "Drug Name"] = "Lestaurtinib"
print(drug_features.shape)
drug_features.head()

(265, 5)


DRUG_ID   Drug_Name                                           Synonyms  \
0        1   Erlotinib  Tarceva, RG-1415, CP-358774, OSI-774, Ro-50823...   
1        3   Rapamycin  AY-22989, Sirolimus, WY-090217, Torisel, Rapamune   
2        5   Sunitinib                 Sutent, Sunitinib Malate, SU-11248   
3        6  PHA-665752                              PHA665752, PHA 665752   
4        9      MG-132                             LLL cpd, MG 132, MG132   

                                Target                     Target_Pathway  
0                                 EGFR                     EGFR signaling  
1                               MTORC1                PI3K/MTOR signaling  
2  PDGFR, KIT, VEGFR, FLT3, RET, CSF1R                      RTK signaling  
3                                  MET                      RTK signaling  
4                    Proteasome, CAPN1  Protein stability and degradation

## Part 1: Get PubChem ids

In [ ]:
#the most recent data on drugs properties
#https://www.cancerrxgene.org/downloads/anova?screening_set=GDSC1
drugs_1 = pd.read_csv(_FOLDER+'drugs_gdsc1.csv')
drug_features = pd.read_csv(_FOLDER + "Drug_Features.csv").rename(columns={"Drug ID": "DRUG_ID", 
                                                                           "Drug Name": "Drug_Name",
                                                                          "Target Pathway": "Target_Pathway"})

drug_names_dict = dict(zip(drug_features["DRUG_ID"], drug_features["Drug_Name"]))

In [ ]:
drugs_2 = pd.read_csv(_FOLDER + 'gdsc2_drug.csv')
drugs_combined = pd.read_csv(_FOLDER + 'drugs_gdsc1_and2.csv')

In [ ]:
drugs_1.head()

drug_id    drug_name               synonyms  \
0     1170   CCT-018159  CCT018159, CCT 018159   
1     1198        FY012                      -   
2     1251  N22899-6-C1                      -   
3     1259  Talazoparib       BMN-673, BMN 973   
4     1359       965-D2                      -   

                        pathway_name                            targets  \
0  Protein stability and degradation                              HSP90   
1                       Unclassified                                  -   
2                       Unclassified  Peptidyl arginine deaminase (PAD)   
3                   Genome integrity                       PARP1, PARP2   
4                     Other, kinases           MKNK1, MKNK2, PIM1, PIM3   

    pubchem  
0   5327091  
1         -  
2         -  
3  44819241  
4         -

In [ ]:
drugs_2.head()

drug_id   drug_name                       synonyms          pathway_name  \
0     1834      743380                              -          Unclassified   
1     1854       MN-64                              -         WNT signaling   
2     1909  Venetoclax  ABT-199, Veneclexta, GDC-0199  Apoptosis regulation   
3     1913    AGI-5198                        IDH-C35            Metabolism   
4     1915     AZD3759                              -        EGFR signaling   

        targets   pubchem  
0             -         -  
1  TNKS1, TNKS2   2802462  
2          BCL2  49846579  
3  IDH1 (R132H)  56645356  
4          EGFR  78209992

In [ ]:
drug_features.head()

DRUG_ID   Drug_Name                                           Synonyms  \
0        1   Erlotinib  Tarceva, RG-1415, CP-358774, OSI-774, Ro-50823...   
1        3   Rapamycin  AY-22989, Sirolimus, WY-090217, Torisel, Rapamune   
2        5   Sunitinib                 Sutent, Sunitinib Malate, SU-11248   
3        6  PHA-665752                              PHA665752, PHA 665752   
4        9      MG-132                             LLL cpd, MG 132, MG132   

                                Target                     Target_Pathway  
0                                 EGFR                     EGFR signaling  
1                               MTORC1                PI3K/MTOR signaling  
2  PDGFR, KIT, VEGFR, FLT3, RET, CSF1R                      RTK signaling  
3                                  MET                      RTK signaling  
4                    Proteasome, CAPN1  Protein stability and degradation

In [ ]:
# drug_ids are not the same in both datasets,
# but we can take mapping of drugname and pubchemid

drugs_1["pubchem"].value_counts().head()

-           137
none         25
several       3
9956119       2
46224516      2
Name: pubchem, dtype: int64

In [ ]:
drugs_2["pubchem"].value_counts().head()

-                     103
None                    5
51042438                2
several                 2
11719003, 58641927      2
Name: pubchem, dtype: int64

In [ ]:
drugs_combined["pubchem"].value_counts().head()

-           240
none         26
several       5
None          5
46883536      3
Name: pubchem, dtype: int64

In [ ]:
# several drug_id can refer to the same drug with the same pubchemid
drugs_1[drugs_1["pubchem"]=="46907787"]

drug_id drug_name        synonyms     pathway_name  \
301     1218       JQ1  JQ-1, (+)-JQ-1  Chromatin other   
380      163       JQ1  JQ-1, (+)-JQ-1  Chromatin other   

                    targets   pubchem  
301  BRD2, BRD3, BRD4, BRDT  46907787  
380  BRD2, BRD3, BRD4, BRDT  46907787

In [ ]:
drugs_2[drugs_2["pubchem"]=="11719003, 58641927"]

drug_id    drug_name            synonyms        pathway_name     targets  \
42      2047  Ulixertinib  BVD-523, VRT752271  ERK MAPK signaling  ERK1, ERK2   
248     1908  Ulixertinib  BVD-523, VRT752271  ERK MAPK signaling  ERK1, ERK2   

                pubchem  
42   11719003, 58641927  
248  11719003, 58641927

In [ ]:
# The GDSC 1 and 2 are different data that has been collected via different lab
# This means it is safe to combine this data so I will use a comnbined data of these

In [65]:
df = drugs_1[(drugs_1["pubchem"]!= "-") & (drugs_1["pubchem"] != "none") & (drugs_1["pubchem"] != "several")]
print("All drugs in gdsc1: %d, With pubchem_ids: %d" % (drugs_1.shape[0], df.shape[0]))

print("Number of drugs in drug_features:", drug_features.shape[0])

# match drug name and pubchem_id based on availble data
gdsc1drug_pubchem_dict = dict(zip(df["drug_name"], df["pubchem"]))
drug_features["pubchem_id"] = drug_features["Drug_Name"].map(gdsc1drug_pubchem_dict)

print("Number of missing data:", drug_features["pubchem_id"].isnull().sum())

# run function calling pubchem to get pubchem ids for drug names
missing_drug_ids = drug_features[drug_features["pubchem_id"].isnull()]["DRUG_ID"].values
pubchem_ids_good, pubchem_ids_bad = get_pubchem_id(missing_drug_ids, drug_names_dict)

All drugs in gdsc1: 403, With pubchem_ids: 238
Number of drugs in drug_features: 265
Number of missing data: 33


100%|██████████| 33/33 [00:34<00:00,  1.04s/it]


In [66]:
# For gdsc1 and gdsc2
df = drugs_combined[(drugs_combined["pubchem"]!= "-") & (drugs_combined["pubchem"] != "none") & (drugs_combined["pubchem"] != "several")]
print("All drugs in gdsc1 and gdsc2: %d, With pubchem_ids: %d" % (drugs_combined.shape[0], df.shape[0]))

print("Number of drugs in drug_features:", drug_features.shape[0])

# match drug name and pubchem_id based on availble data
drug_pubchem_dict = dict(zip(df["drug_name"], df["pubchem"]))
drug_features["pubchem_id"] = drug_features["Drug_Name"].map(drug_pubchem_dict)

print("Number of missing data:", drug_features["pubchem_id"].isnull().sum())

# run function calling pubchem to get pubchem ids for drug names
missing_drug_ids = drug_features[drug_features["pubchem_id"].isnull()]["DRUG_ID"].values
pubchem_ids_good, pubchem_ids_bad = get_pubchem_id(missing_drug_ids, drug_names_dict)

All drugs in gdsc1 and gdsc2: 700, With pubchem_ids: 429
Number of drugs in drug_features: 265
Number of missing data: 31


100%|██████████| 31/31 [00:31<00:00,  1.02s/it]


In [67]:
# run corrections
print("Number of found ids %d, Number of not found: %d" % (len(pubchem_ids_good), len(pubchem_ids_bad)))

if "DRUG_ID" in drug_features.columns:
    drug_features.set_index("DRUG_ID", inplace = True)

for drug_id in pubchem_ids_good:
    drug_features.loc[drug_id, "pubchem_id"] = pubchem_ids_good[drug_id]

Number of found ids 6, Number of not found: 25


In [68]:
pubchem_ids_good, not_identified_drugs = run_manual_corrections(pubchem_ids_bad, drop_not_found_drugs = False)

for drug_id in pubchem_ids_good:
    drug_features.loc[drug_id, "pubchem_id"] = pubchem_ids_good[drug_id]
    
print("\nNumber of found ids found after manual corrections %d, Number of not found: %d" % (len(pubchem_ids_good), len(not_identified_drugs)))

Total number of drugs: 25
Total number of drugs for correction: 25
Number of corrected drugs: 10
Number of not found drugs: 15
Final number of drugs with PubChem id: 25

Number of found ids found after manual corrections 10, Number of not found: 15


In [69]:
drug_features[drug_features["pubchem_id"].isnull()]

Drug_Name Synonyms                                 Target  \
DRUG_ID                                                                     
164                  JQ12        -                           HDAC1, HDAC2   
211              TL-2-105        -                            not defined   
225      Genentech Cpd 10        -                           AURKA, AURKB   
253              XMD14-99        -                 ALK, CDK7, LTK, others   
261               TL-1-85        -                                    TAK   
286            KIN001-236        -                Angiopoietin-1 receptor   
330               XMD13-2        -                                  RIPK1   
332              XMD15-27        -                                  CAMK2   
1037                BX796   BX-796  TBK1, PDK1 (PDPK1), IKK, AURKB, AURKC   
1142          HG-5-113-01        -             LOK, LTK, TRCB, ABL(T315I)   
1143           HG-5-88-01        -                            EGFR, ADCK4   
1158            XMD11-85h        -  BRSK2, FLT4, MARK4, PRKCD, RET, SPRK1   
1166           QL-VIII-58        -                              MTOR, ATR   
1203            QL-XII-61        -                               BMX, BTK   
1261               rTRAIL        -                 TRAIL receptor agonist   

                        Target_Pathway pubchem_id  
DRUG_ID                                            
164      Chromatin histone acetylation        NaN  
211                 ERK MAPK signaling        NaN  
225                            Mitosis        NaN  
253                              Other        NaN  
261                     Other, kinases        NaN  
286                              Other        NaN  
330               Apoptosis regulation        NaN  
332                     Other, kinases        NaN  
1037                             Other        NaN  
1142                             Other        NaN  
1143                    Other, kinases        NaN  
1158                             Other        NaN  
1166                             Other        NaN  
1203                    Other, kinases        NaN  
1261              Apoptosis regulation        NaN

In [70]:
drug_features = drug_features[~drug_features["pubchem_id"].isnull()]
drug_features.shape

(250, 5)

**15 drugs from 265 can't be assesed and will not be used in further modelling**

## Part 2 :Getting properties by PubChem API

In [71]:
%%time
drug_features, drugs_to_drop = get_pubchem_properties(drug_features)

100%|██████████| 250/250 [04:52<00:00,  1.17s/it]

CPU times: user 2.88 s, sys: 395 ms, total: 3.27 s
Wall time: 4min 52s


In [72]:
len(drugs_to_drop)

0

In [73]:
drug_features.drop(drugs_to_drop, axis=0).to_csv(_FOLDER_2 + "drug_features_raw.csv")

## End of Part 2: Read prepared data

In [74]:
drug_features = pd.read_csv(_FOLDER_2+ "drug_features_raw.csv").set_index("DRUG_ID")
print(drug_features.shape)

(250, 26)


In [75]:
# pubchem_id is none

drug_features[drug_features["molecular_weight"].isnull()].shape[0]

0

In [76]:
drug_features.columns

Index(['Drug_Name', 'Synonyms', 'Target', 'Target_Pathway', 'pubchem_id',
       'molecular_weight', 'elements', '2bonds', '3bonds', 'xlogp',
       'formal_charge', 'surface_area', 'complexity', 'h_bond_donor_count',
       'h_bond_acceptor_count', 'rotatable_bond_count', 'heavy_atom_count',
       'atom_stereo_count', 'defined_atom_stereo_count',
       'undefined_atom_stereo_count', 'bond_stereo_count',
       'covalent_unit_count', 'molecular_formula', 'canonical_smiles',
       'inchi_string', 'inchi_key'],
      dtype='object')

In [77]:
int_columns = ['2bonds', '3bonds', 'h_bond_donor_count',
       'h_bond_acceptor_count', 'rotatable_bond_count', 'heavy_atom_count',
       'atom_stereo_count', 'defined_atom_stereo_count',
       'undefined_atom_stereo_count', 'bond_stereo_count',
       'covalent_unit_count']
for col in int_columns:
    drug_features[col] = np.int16(drug_features[col])

## Part 3: Preprocessing Text PubChem characteristics

### Presence of some elements (11 elements)

In [78]:
%%time

all_elements = list(set(drug_features["elements"].str.split(",", expand=True).fillna(0).values.flatten())- set([0," 'C'", "'C'", " 'H'"]))
all_elements

elements_in_drugs= list(set([atom.strip(" ").strip("'") for atom in all_elements]))
exceptions =[]
for drug_index in drug_features.index:
    compound_elements = drug_features.loc[drug_index, "elements"]
#     print(compound_elements)
    try:
        for i, atom in list(enumerate(elements_in_drugs)):
            if atom in compound_elements:
                drug_features.loc[drug_index, atom] = 1
#                 print(atom, "Yes")
            else:
                drug_features.loc[drug_index, atom] = 0
#                 print(atom, "No")
    except:
        exceptions.append(drug_index)
        drug_features.loc[drug_index, atom] = 0
        
for col in ['B', 'I', 'Br', 'Cl', 'O', 'N', 'F', 'P', 'S', 'Pt']:
    drug_features[col]= np.int16(drug_features[col])
        
print("Exceptions:", drug_features.loc[exceptions, :].shape[0])
print("Elements in drugs:", len(elements_in_drugs), elements_in_drugs)

Exceptions: 0
Elements in drugs: 10 ['Br', 'S', 'Pt', 'Cl', 'P', 'O', 'B', 'N', 'F', 'I']
CPU times: user 159 ms, sys: 557 µs, total: 159 ms
Wall time: 159 ms


In [ ]:
drug_features["Br"].value_counts()

0    243
1      7
Name: Br, dtype: int64

In [ ]:
drug_features.to_csv(_FOLDER_2 + "drug_features_with_pubchem_properties.csv")

PubChem_features = ["molecular_weight","2bonds", "3bonds", "xlogp", "formal_charge", 
    "surface_area", "complexity", "h_bond_donor_count", 
    "h_bond_acceptor_count", "rotatable_bond_count",
    "heavy_atom_count", "atom_stereo_count", "defined_atom_stereo_count",
    "undefined_atom_stereo_count", "bond_stereo_count", "covalent_unit_count",
    'B', 'I', 'Br', 'Cl', 'O', 'N', 'F', 'P', 'S', 'Pt']

with open(_FOLDER_2 + "X_PubChem_properties.txt", 'w') as f:
    for s in PubChem_features:
        f.write(str(s) + '\n')

print("Number of PubChem features:", len(PubChem_features))

Number of PubChem features: 26


## End of Part 3: Read Prepared Data

In [ ]:
drug_features = pd.read_csv(_FOLDER_2+ "drug_features_with_pubchem_properties.csv").set_index("DRUG_ID")

with open(_FOLDER_2 + "X_PubChem_properties.txt", 'r') as f:
    X_PubChem_properties = [line.rstrip('\n') for line in f]

## Part 4: Preprocessing Drugs description from original data

In this section, we are going to have some dumnies columns for Target and Target_Pathway

Converting of Target Pathway resulted in 26 new columns

It is also worth considering elements columns and that deleting columns with C and H which are present in all the compounds

### Dumnies for Target (229) and Target_Pathway (23)

In [ ]:
drug_features.head(3)

Drug_Name                                           Synonyms  \
DRUG_ID                                                                 
1        Erlotinib  Tarceva, RG-1415, CP-358774, OSI-774, Ro-50823...   
3        Rapamycin  AY-22989, Sirolimus, WY-090217, Torisel, Rapamune   
5        Sunitinib                 Sutent, Sunitinib Malate, SU-11248   

                                      Target       Target_Pathway  pubchem_id  \
DRUG_ID                                                                         
1                                       EGFR       EGFR signaling      176870   
3                                     MTORC1  PI3K/MTOR signaling     5384616   
5        PDGFR, KIT, VEGFR, FLT3, RET, CSF1R        RTK signaling     5329102   

         molecular_weight                 elements  2bonds  3bonds  xlogp  \
DRUG_ID                                                                     
1                  393.40       'C', 'H', 'O', 'N'       8       1    3.3   
3                  184.18            'C', 'H', 'N'       4       2    1.3   
5                  398.50  'C', 'N', 'H', 'O', 'F'       8       0    2.6   

         ...  N  Cl  I    H  P  O  Br  F  S  B  
DRUG_ID  ...                                    
1        ...  1   0  0  1.0  0  1   0  0  0  0  
3        ...  1   0  0  1.0  0  0   0  0  0  0  
5        ...  1   0  0  1.0  0  1   0  1  0  0  

[3 rows x 37 columns]

In [ ]:
targets = ""
for x in drug_features["Target"].values:
    targets = targets + ", " + x
targets = list(set(targets.split(", ")[1:]))
print("Number of targets:", len(targets))

Number of targets: 212


In [ ]:
df_target = pd.DataFrame(data = np.int32(np.zeros([drug_features.shape[0], len(targets)])), 
                         index = drug_features.index, 
                         columns = targets)
len(targets)

212

In [ ]:
for index in drug_features.index:
    targets_i = drug_features.loc[index, "Target"].split(", ")
    df_target.loc[index, targets_i]=1
df_target.shape

(250, 213)

In [ ]:
print("Number of unique pathways:", drug_features["Target_Pathway"].nunique())

df_target_target_pathway = pd.concat([df_target, pd.get_dummies(drug_features["Target_Pathway"])], axis=1)
df_target_target_pathway.shape

Number of unique pathways: 23


(250, 236)

In [ ]:
drug_features["Target_Pathway"]

DRUG_ID
1                          EGFR signaling
3                     PI3K/MTOR signaling
5                           RTK signaling
6                           RTK signaling
9       Protein stability and degradation
                      ...                
1498                   ERK MAPK signaling
1502                      Hormone-related
1526                   ERK MAPK signaling
1527                  PI3K/MTOR signaling
1529                                Other
Name: Target_Pathway, Length: 250, dtype: object

In [ ]:
df_final = pd.concat([drug_features.drop(["Target_Pathway"], axis=1), df_target_target_pathway], axis=1)
df_final.shape

(250, 272)

In [ ]:
df_final

Drug_Name                                           Synonyms  \
DRUG_ID                                                                    
1           Erlotinib  Tarceva, RG-1415, CP-358774, OSI-774, Ro-50823...   
3           Rapamycin  AY-22989, Sirolimus, WY-090217, Torisel, Rapamune   
5           Sunitinib                 Sutent, Sunitinib Malate, SU-11248   
6          PHA-665752                              PHA665752, PHA 665752   
9              MG-132                             LLL cpd, MG 132, MG132   
...               ...                                                ...   
1498      Selumetinib                        AZD6244, AZD-6244, ARRY-886   
1502     Bicalutamide           ICI-176334, Casodex, Cosudex, ICI 176334   
1526      Refametinib                   RDEA119, BAY-86-9766, BAY 869766   
1527       Pictilisib                         GDC-0941, GDC0941, RG-7621   
1529     Pevonedistat                        MLN4924, MLN 4924, MLN-4924   

                                      Target  pubchem_id  molecular_weight  \
DRUG_ID                                                                      
1                                       EGFR      176870            393.40   
3                                     MTORC1     5384616            184.18   
5        PDGFR, KIT, VEGFR, FLT3, RET, CSF1R     5329102            398.50   
6                                        MET    10461815            641.60   
9                          Proteasome, CAPN1      462382            475.60   
...                                      ...         ...               ...   
1498                              MEK1, MEK2    10127622            457.70   
1502                                      AR        2375            430.40   
1526                              MEK1, MEK2    44182295            572.30   
1527                          PI3K (class 1)    17755052            513.60   
1529                                     NAE    16720766            443.50   

                                    elements  2bonds  3bonds  xlogp  \
DRUG_ID                                                               
1                         'C', 'H', 'O', 'N'       8       1    3.3   
3                              'C', 'H', 'N'       4       2    1.3   
5                    'C', 'N', 'H', 'O', 'F'       8       0    2.6   
6              'C', 'N', 'Cl', 'H', 'O', 'S'      13       0    5.0   
9                         'C', 'H', 'O', 'N'       7       0    4.8   
...                                      ...     ...     ...    ...   
1498     'C', 'N', 'Cl', 'H', 'O', 'Br', 'F'       8       0    3.6   
1502            'C', 'N', 'H', 'O', 'F', 'S'       9       1    2.3   
1526       'C', 'I', 'N', 'H', 'O', 'F', 'S'       8       0    2.8   
1527                 'C', 'N', 'H', 'O', 'S'      10       0    1.6   
1529                 'C', 'N', 'H', 'O', 'S'       9       0    1.7   

         formal_charge  ...  JNK and p38 signaling  Metabolism  Mitosis  \
DRUG_ID                 ...                                               
1                  0.0  ...                      0           0        0   
3                  0.0  ...                      0           0        0   
5                  0.0  ...                      0           0        0   
6                  0.0  ...                      0           0        0   
9                  0.0  ...                      0           0        0   
...                ...  ...                    ...         ...      ...   
1498               0.0  ...                      0           0        0   
1502               0.0  ...                      0           0        0   
1526               0.0  ...                      0           0        0   
1527               0.0  ...                      0           0        0   
1529               0.0  ...                      0           0        0   

         Other  Other, kinases  PI3K/MTOR signaling  \
DRUG_ID                                              

In [ ]:
df_final.to_csv(_FOLDER_2 + "drug_features_with_pubchem_properties_final.csv")

with open(_FOLDER_2 + "X_features_Targets.txt", 'w') as f:
    for s in targets:
        f.write(str(s) + '\n')
        
with open(_FOLDER_2 + "X_features_Target_Pathway.txt", 'w') as f:
    for s in drug_features["Target_Pathway"].unique():
        f.write(str(s) + '\n')   

## End of Part 4: Read Prepared Data

In [ ]:
drug_features = pd.read_csv(_FOLDER_2 + "drug_features_with_pubchem_properties_final.csv")

with open(_FOLDER_2 + "X_features_Targets.txt", 'r') as f:
    X_features_Targets = [line.rstrip('\n') for line in f]
    
with open(_FOLDER_2 + "X_features_Target_Pathway.txt", 'r') as f:
    X_features_Target_Pathway = [line.rstrip('\n') for line in f]

In [ ]:
drug_features.head()

DRUG_ID   Drug_Name                                           Synonyms  \
0        1   Erlotinib  Tarceva, RG-1415, CP-358774, OSI-774, Ro-50823...   
1        3   Rapamycin  AY-22989, Sirolimus, WY-090217, Torisel, Rapamune   
2        5   Sunitinib                 Sutent, Sunitinib Malate, SU-11248   
3        6  PHA-665752                              PHA665752, PHA 665752   
4        9      MG-132                             LLL cpd, MG 132, MG132   

                                Target  pubchem_id  molecular_weight  \
0                                 EGFR      176870            393.40   
1                               MTORC1     5384616            184.18   
2  PDGFR, KIT, VEGFR, FLT3, RET, CSF1R     5329102            398.50   
3                                  MET    10461815            641.60   
4                    Proteasome, CAPN1      462382            475.60   

                        elements  2bonds  3bonds  xlogp  ...  \
0             'C', 'H', 'O', 'N'       8       1    3.3  ...   
1                  'C', 'H', 'N'       4       2    1.3  ...   
2        'C', 'N', 'H', 'O', 'F'       8       0    2.6  ...   
3  'C', 'N', 'Cl', 'H', 'O', 'S'      13       0    5.0  ...   
4             'C', 'H', 'O', 'N'       7       0    4.8  ...   

   JNK and p38 signaling  Metabolism  Mitosis  Other  Other, kinases  \
0                      0           0        0      0               0   
1                      0           0        0      0               0   
2                      0           0        0      0               0   
3                      0           0        0      0               0   
4                      0           0        0      0               0   

   PI3K/MTOR signaling  Protein stability and degradation  RTK signaling  \
0                    0                                  0              0   
1                    1                                  0              0   
2                    0                                  0              1   
3                    0                                  0              1   
4                    0                                  1              0   

   WNT signaling  p53 pathway  
0              0            0  
1              0            0  
2              0            0  
3              0            0  
4              0            0  

[5 rows x 273 columns]

In [ ]:
with open(_FOLDER_2+"X_features_cancer_cell_lines.txt", 'r') as f:
    X_cancer_cell_lines = [line.rstrip('\n') for line in f]

In [ ]:
print("Final Features: \n")
print("Cell lines (CCL) features:", len(X_cancer_cell_lines))
print("PubChem drug features:", len(PubChem_features))
print("Drug description features - Targets: %d, Target_Pathway: %d" % (len(X_features_Targets), len(X_features_Target_Pathway)))

Final Features: 

Cell lines (CCL) features: 1073
PubChem drug features: 26
Drug description features - Targets: 212, Target_Pathway: 23


In [ ]:
all_elements

["'F'",
 " 'S'",
 " 'N'",
 " 'Cl'",
 " 'B'",
 "'H'",
 " 'Br'",
 " 'P'",
 " 'I'",
 " 'O'",
 " 'F'",
 " 'Pt'"]